lib/internal/streams/readable.js

```diff
@@ -1023,53 +1116,57 @@ function nReadingNextTick(self) {
 // If the user uses them, then switch into old mode.
 Readable.prototype.resume = function() {
   const state = this._readableState;
-  if (!state.flowing) {
+  if ((state.state & kFlowing) === 0) {
     debug('resume');
     // We flow only if there is no one listening
     // for readable, but we still have to call
     // resume().
-    state.flowing = !state.readableListening;
+    state.state |= kHasFlowing;
+    if ((state.state & kReadableListening) === 0) {
+      state.state |= kFlowing;
+    } else {
+      state.state &= ~kFlowing;
+    }
     resume(this, state);
   }
-  state[kPaused] = false;
+  state.state |= kHasPaused
+  state.state &= ~kPaused;
   return this;
 };
 
 function resume(stream, state) {
-  if (!state.resumeScheduled) {
-    state.resumeScheduled = true;
+  if ((state.state & kResumeScheduled) === 0) {
+    state.state |= kResumeScheduled;
     process.nextTick(resume_, stream, state);
   }
 }
 
 function resume_(stream, state) {
-  debug('resume', state.reading);
-  if (!state.reading) {
+  if ((state.state & kReading) === 0) {
     stream.read(0);
   }
 
-  state.resumeScheduled = false;
+  state.state &= ~kResumeScheduled;
   stream.emit('resume');
   flow(stream);
-  if (state.flowing && !state.reading)
+  if ((state.state & (kFlowing & kReading)) === kFlowing) {
     stream.read(0);
+  }
 }
 
 Readable.prototype.pause = function() {
-  debug('call pause flowing=%j', this._readableState.flowing);
-  if (this._readableState.flowing !== false) {
-    debug('pause');
-    this._readableState.flowing = false;
+  const state = this._readableState;
+  if ((state.state & (kHasFlowing | kFlowing)) !== kHasFlowing) {
+    state.state &= ~kFlowing;
     this.emit('pause');
   }
-  this._readableState[kPaused] = true;
+  state.state |= kHasPaused | kPaused;
   return this;
 };
 
 function flow(stream) {
   const state = stream._readableState;
-  debug('flow', state.flowing);
```

Comment:
Do we care about these debug statements btw? Does anyone still use them?

Reply:
I'd say they're useful when debugging streams.

Reply:
Never use them

Reply:
@mscdex have you used them in the last let's say 5 years?

Reply:
For sure, even specifically when dealing with a bug in node's stream implementation.

Reply:
Making unrelated changes like this should be a part of a separate PR anyway, but I think it's worth keeping. The performance impact of them should be minimized already since we swap them out with a noop function if debugging is not enabled.

Reply:
I mostly removed them
Because the changes of this or make it even more verbose and more noisy than it already was.